File: javascript/Filter.js
const fs = require('fs');

class Filter {
    constructor() {
        this.hardFile = "../hard.txt";
        this.hardRegex = "";
        this.softFile = "../soft.txt";
        this.softRegex = "";
        this.dictionary = {};
        this.text = "";

        this.initDictionary();
        this.initRegex();
    }

    initDictionary() {
        const getResourceData = function (file) {
            return fs.readFileSync(file, 'utf8').split("\n").join("|");
        };

        this.dictionary = {
            'hard': getResourceData(this.hardFile),
            'soft': getResourceData(this.softFile),
        };

        return this;
    }

    initRegex() {
        this.setHardRegex(new RegExp("(" + this.dictionary.hard + ")", "ui"));
        this.setSoftRegex(new RegExp("(\\b)+(" + this.dictionary.soft + ")+(\\b)", "gui"));

        return this;
    }

    setHardFile (file) {
        this.hardFile = file;

        return this;
    }

    setSoftFile (file) {
        this.softFile = file;

        return this;
    }

    setDictionary (dictionary) {
        this.dictionary = dictionary;

        return this;
    }

    setText(text) {
        this.text = text;

        return this;
    }

    setHardRegex(regex) {
        this.hardRegex = regex;

        return this;
    }

    setSoftRegex(regex) {
        this.softRegex = regex;

        return this;
    }

    checkHard() {
        return this.text.match(this.hardRegex);
    }

    checkSoft() {
        return this.text.match(this.softRegex);
    }

    replaceHard (replaceText) {
        return this.text.replace(this.hardRegex, replaceText);
    }

    replaceSoft (replaceText) {
        return this.text.replace(this.softRegex, replaceText);
    }

    replace (replaceText) {
        const softText = this.replaceSoft(replaceText);
        
        return this.setText(softText).replaceHard(replaceText);
    }
}


module.exports = Filter;

File: python/filter.py
import re
import random


class Filter:
    hardFile = "../hard.txt"
    hardRegex = ""
    softFile = "../soft.txt"
    softRegex = ""
    dictionary = {}
    text = ""

    def __init__(self):
        self.init_dictionary()
        self.init_regex()

    def init_dictionary(self):
        with open(self.hardFile, 'r') as lines:
            self.dictionary['hard'] = "|".join([line.rstrip('\n') for line in lines])

        with open(self.softFile, 'r') as lines:
            self.dictionary['soft'] = "|".join([line.rstrip('\n') for line in lines])

        return self

    def init_regex(self):
        self.hardRegex = r"(" + self.dictionary['hard'] + ")"
        self.softRegex = r"(\b)+({})+(\b)".format(self.dictionary['soft'])

    def set_hard_file(self, file):
        self.hardFile = file
        return self

    def set_hard_regex(self, regex):
        self.hardRegex = regex
        return self

    def set_soft_file(self, file):
        self.softFile = file
        return self

    def set_soft_regex(self, regex):
        self.softRegex = regex
        return self

    def set_dictionary(self, dictionary):
        self.dictionary = dictionary
        return self

    def set_text(self, text):
        self.text = text
        return self

    def check_hard(self):
        matches = re.search(self.hardRegex, self.text, re.IGNORECASE | re.UNICODE)

        if matches:
            return True

        return False

    def check_soft(self):
        matches = re.search(self.softRegex, self.text, re.IGNORECASE | re.UNICODE)

        if matches:
            return True

        return False

    def replace_hard(self, replace_text):
        return re.sub(self.hardRegex, replace_text, self.text)

    def replace_soft(self, replace_text):
        return re.sub(self.softRegex, replace_text, self.text)

    def replace(self, replace_text):
        soft_text = self.replace_soft(replace_text)

        self.set_text(soft_text)

        return self.replace_hard(replace_text)


emojis = ["🤐", "😡", "😤", "😠"]

app = Filter()
text = app\
    .set_hard_file("../hard.txt")\
    .set_soft_file("../soft.txt")\
    .init_dictionary()\
    .set_text("yarak kürek işler bunlar amk tuzlayarak ansiklopedi") \
    .replace(random.choice(emojis))

print(text)